src/rust/ide/span-tree/src/generate.rs

```diff
@@ -73,29 +55,35 @@ impl ChildGenerator {
     }
 
     fn generate_ast_node
-    (&mut self, child_ast:Located<Ast>, ctx:Context)
-    -> FallibleResult<&node::Child> {
-        let child = node::Child {
-            node                : child_ast.item.generate_node(ctx)?,
-            offset              : self.current_offset,
-            chained_with_parent : ast_can_be_chained_with_parent(&child_ast,ctx),
-            ast_crumbs          : child_ast.crumbs
-        };
+    (&mut self, child_ast:Located<Ast>, kind:node::Kind) -> FallibleResult<&node::Child> {
+        let node = child_ast.item.generate_node(kind)?;
+        Ok(self.add_node(child_ast.crumbs,node))
+    }
+
+    fn add_node(&mut self, ast_crumbs:ast::Crumbs, node:Node) -> &node::Child {
+        let offset = self.current_offset;
+        let child = node::Child {node,ast_crumbs,offset};
         self.current_offset += child.node.size;
         self.children.push(child);
-        Ok(self.children.last().unwrap())
+        self.children.last().unwrap()
     }
 
-    fn generate_empty_node(&mut self) -> &node::Child {
+    fn generate_empty_node(&mut self, insert_type:InsertType) -> &node::Child {
         let child = node::Child {
-            node                : Node::new_empty(),
+            node                : Node::new_empty(insert_type),
             offset              : self.current_offset,
-            chained_with_parent : false,
             ast_crumbs          : vec![]
         };
         self.children.push(child);
         self.children.last().unwrap()
     }
+
+    fn reverse_children(&mut self) {
+        self.children.reverse();
+        for child in &mut self.children {
+            child.offset = self.current_offset - child.offset - child.node.size;
```

Comment:
Is it impossible to trigger usize wraparound?

Reply:
Yes, it is